### DISCUSSION TITLE
directory named structured

### DISCUSSION THREAD

**Discussion by jhcao23** [danny-avila/LibreChat#593] [Q&A]
Quick question - we have a dir named `structured` inside `api/app/langchain/tools/`. Its files seems to be very similar to those outside it such as `StableDiffusion.js`, `Wolfram.js`. What's this `structured` dir used for?

**Comment by danny-avila** [comment]
For improved tool handling through OpenAI functions. This allows you to call tools with multiple parameters instead of one. any tools with a structured counterpart will be used when using the functions agent instead of their regular Tool class.

I still need to update the documentation on this.
